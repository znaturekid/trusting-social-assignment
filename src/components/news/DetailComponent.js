import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { formatDate, getFirstImageLink } from '../../utils';
import PropTypes from 'prop-types';
const DetailWrapper = styled.div`
    color: #000;
    display: grid;
    grid-gap: 10px;
    justify-content: center;
    flex-direction: column;
    padding: 40px;
    @media screen and (max-width: 600px) {
        padding: 20px 0;
    }
`

const Snippet = styled.p`

`
const PageEmpty = styled.div`
    display: flex;
    height: 80vh;
    justify-content: center;
    align-items: center;
    font-size: 20px;
    flex-direction: column;
`
const P = styled.p`
    background: #ccc;
    padding: 5px;
    font-size: 13px;
    margin-bottom: 5px;
`

const Media = styled.img`
    width: 100%;
`

class DetailComponent extends React.Component {
    
    render () {
        const { detail } = this.props;
        if (!detail) {
            return (<PageEmpty>
                There are no content.
                <Link to='/'>Go back Home Page!</Link>
            </PageEmpty>)
        }
        const src = getFirstImageLink(detail.multimedia);
        return (<DetailWrapper>
            <P>From {detail.source} on {formatDate(detail.pub_date)}</P>
            {src && <Media src={src} alt=""></Media>}
            <Snippet>{detail.snippet}</Snippet>
        </DetailWrapper>)
    }
}

DetailComponent.propTypes = {
    detail: PropTypes.object
}

export default DetailComponent;