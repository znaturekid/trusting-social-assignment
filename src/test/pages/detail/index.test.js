import React from 'react';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DetailComponent from '../../../pages/detail';
import { mountWrap, shallowWrap } from '../../help';
import configureMockStore from 'redux-mock-store';

import thunk from 'redux-thunk';
import { Provider } from 'react-redux';;

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

Enzyme.configure({
    adapter: new Adapter()
});

const store = mockStore({
    news: {
        newLists: [],
        page: 1,
        query: '',
        loading: false
    }
})
const props = {
    
}
const component = (<Provider store={store}><DetailComponent></DetailComponent></Provider>)

describe('components', () => {
    
    describe('Detail =>', () => {
        it('should render detail correctly', () => {
            const detail = shallowWrap(component);
            expect(detail).toMatchSnapshot();
        })
    })
})