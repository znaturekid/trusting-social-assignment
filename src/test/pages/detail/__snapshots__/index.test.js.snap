// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Detail => should render detail correctly 1`] = `<DetailPage />`;

exports[`components Home => should render home correctly 1`] = `<DetailPage />`;
